nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -588,6 +589,63 @@ def modify_state_dict(self, conf, state_dict):
                 new_state_dict[new_key] = state_dict[key]
             state_dict = new_state_dict
 
+        # compatibility for inductor in inference
+        if not conf.get('inductor', False):
```

Comment:
All of this logic should be in a function separate from the core function here of modify state dict. Call that function inside here